pkg/controller/runtimecomponent/runtimecomponent_controller.go

```diff
@@ -599,6 +601,25 @@ func (r *ReconcileRuntimeComponent) Reconcile(request reconcile.Request) (reconc
 		}
 	} else {
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", routev1.SchemeGroupVersion.String()))
+		if instance.Spec.Expose != nil && *instance.Spec.Expose {
+			ing := &networkingv1beta1.Ingress{ObjectMeta: defaultMeta}
+			err = r.CreateOrUpdate(ing, instance, func() error {
+				appstacksutils.CustomizeIngress(ing, instance)
+				return nil
+			})
+			if err != nil {
+				reqLogger.Error(err, "Failed to reconcile Ingress")
+				return r.ManageError(err, common.StatusConditionTypeReconciled, instance)
+			}
+		} else {
+			ing := &networkingv1beta1.Ingress{ObjectMeta: defaultMeta}
+			err = r.DeleteResource(ing)
```

Comment:
For the Knative scenario, please delete the `ingress` resource

Reply:
Good catch